### PR TITLE
Turn Send screen into a functional component

### PR DESCRIPTION
**Pull request by gnardini**
### Description

This PR is part 1 of https://github.com/celo-org/wallet/pull/406

This PR refactors the Send screen to be a functional component rather than a class component to make it easier to review the actual changes in the other PR. Unfortunately, the nature of the changes make reviewing the Send screen specifically pretty difficult. The only real things that changed here are:
- Added SendHeader to replace the react-navigation provided header (more on this below).
- Re-added the verification banner on the Send screen that we removed during the CB earn campaign (and deleted the TODOs). Not hiding the contacts though for now.

#### The Send screen header
The Send screen now has a custom header. Created a `CustomHeader` component for this. The reason is that using the header provided by react-navigation like we've been doing causes a bunch of navigation issues. The screen has an animation where it shows up from the bottom. To achieve that we needed to put the Send screen in a weird place in the navigation tree because otherwise the header comes up from the top instead. Since the screen wasn't in the stack navigation object, navigating from the Send screen to another screen caused the send screen to disappear from the stack, so if the user navigated to somewhere else (SendAmount or on a CTA in the list header) and back the user actually returned to the Home screen.
Having this custom header fixes these issues without negative consequences that I could find.

### Other changes

N/A

### Tested

Manually mostly.

### How others should test

- Open the app, tap the `Send` button.
- Check that the list of contacts works well, it filters the contacts correctly when typing in the search input.
- Go to the next screen (`SendAmount`) by picking a contact and back. See that you're back on the Send screen.

### Related issues

- Part of #405

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/send/Send.tsx [hunk 1]
```diff
@@ -282,33 +151,26 @@ class Send extends React.Component<Props, State> {
     return sections
   }
 
-  renderListHeader = () => {
-    // CB TEMPORARY HOTFIX: Disabling phone number confirmation requirement
-    // for sending to phone numbers
-    // const { t, numberVerified, verificationPossible, inviteRewardsEnabled } = this.props
```
Comment (jeanregisser) [inline packages/mobile/src/send/Send.tsx:288]: Just to confirm, we really want to restore the original behavior?

**Review by jeanregisser — APPROVED** [review]
This is great! Thanks! 👍

**Comment by Celoqa** [comment]
Hi @tarikbellamine We have verified this using latest Android internal build V1.15.0 & Testflight build .15.0(63) and observe the followings:

- [ ] User Redirected to send page when tapping on Send button in home page.

- [ ] User is able to see "Sending to friend?" banner on Send page if user not confirmed phone number 

- [x] User is able to see the Saved contact phone number in the same format they are saved in,(EX: saved with country code) [Attachment1](https://drive.google.com/file/d/1AVwRVQLuFEf-M08-HQtxXkkoYoqkKU6z/view?usp=sharing) (EX: saved without country code) [Attachment2](https://drive.google.com/file/d/1dXAafnqZrxNiJXeg3wFiVVE9CJUyDxm9/view?usp=sharing)

- [ ] User able to see **Send to Phone number** to saved contact when user search with phone number instead of saved contact. [Attachment](https://drive.google.com/file/d/1TYn4otf6T91KVLI-1-_7ETYQcyHiDAAE/view?usp=sharing) 

- [ ]  User not able to see the saved contact by using phone number instead of Country code + phone number if user saved the contact with country code.[Attachment](https://drive.google.com/file/d/1astnH4uiaZvlunda6HHlVz-NtuabrWq0/view?usp=sharing)
- [ ] User redirected to send amount page after tapping on searched contact.
- [ ] User redirected back to send page by tapping on < button on send amount page.

Can you please let us know we need to open a new issue for uncheck scenario OR it is expected?

**Comment by MuckT** [comment]
https://valora.testquality.com requirement [R24](https://valora.testquality.com/project/12128/testcase/286102/requirement/11357). Do not remove or edit.

**Comment by tarikbellamine** [comment]
[Issue](https://github.com/celo-org/wallet/issues/475) opened for the above. Thank you!

**Pull request merged**